spec/vra_spec.rb

```diff
@@ -102,14 +103,34 @@
       expect(state[:resource_id]).to eq('e8706351-cf4c-4c12-acb7-c90cc683b22c')
     end
 
-    describe 'getting the IP address from the server' do
+    describe 'getting the IP address from the server when use_dns is false' do
       context 'when no IP addresses are returned' do
         it 'raises an exception' do
           allow(resource).to receive(:ip_addresses).and_return([])
           expect { driver.create(state) }.to raise_error(RuntimeError)
         end
       end
+      context 'when IP addresses are returned' do
+        it 'sets the IP address as the hostname in the state hash' do
+          driver.create(state)
+          expect(state[:hostname]).to eq('1.2.3.4')
+        end
+      end
+    end
 
+    describe 'getting the IP address from the server when use_dns is true' do
+      context 'when no IP addresses are returned' do
+        let(:config) do
```

Comment:
We generally reserve `do ... end` blocks for items where multiple lines are required.  Since you only have one key-value pair, we can probably shrink this down to a single line:

```ruby
let(:config) { { use_dns: true } }
```